xbmc/storage/discs/AbstractDiscDriveHandler.h

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+#pragma once
+
+#include "IDiscDriveHandler.h"
+
+#include <string>
+
+/*! \brief Implementation of an abstract DiscDriveHandler
+* \sa IDiscDriveHandler
+*/
+class AbstractDiscDriveHandler : public IDiscDriveHandler
+{
+public:
+  /*! \brief Get the optical drive state provided its device path
+  * \sa IDiscDriveHandler
+  * \note Subclasses must implement this method, AbstractDiscDriveHandler itself will throw an exception
+  * \param devicePath the path for the device drive (e.g. /dev/sr0)
+  * \return The drive state
+  */
+  virtual DriveState GetDriveState(const std::string& devicePath) = 0;
```

Comment:
IDiscDriveHandler already declares these metahods as pure virtual. So, no need to do this here a second time?